### ISSUE TITLE
Cluster ready count not updating in git repo when I move cluster to new workspace

### ISSUE THREAD

**Issue by tfiduccia** [rancher/dashboard#1632]
Steps:
1.  Create a git repo and target one of your clusters
2.  Wait for it to complete
3.  Move your cluster to another workspace

Results:  Clusters Ready count on Git Repos list page not updating.  Updates on workspaces & Cluster group list pages.
![image](https://user-images.githubusercontent.com/11514927/95144556-cda50980-072d-11eb-8f0a-fe2a0b5f351a.png)

**Comment by aalves08** [comment]
going to fleet -> Clusters (list) -> hit 3 dots on the cluster item -> Choose assign to -> Select another workspace

Request seems to go down fine:
```
Request URL: https://localhost:8005/v3/clusters/c-m-l8bdcq5f
Request Method: PUT
Status Code: 200 OK
```
![Screenshot 2022-02-16 at 16.50.39.png](https://zube.io/files/rancher/1db8374c2a50bf808221891aeabbc9cd-screenshot-2022-02-16-at-16-50-39.png)
![Screenshot 2022-02-16 at 16.51.01.png](https://zube.io/files/rancher/3e5e517634fa9ac672c798bc8b4569bc-screenshot-2022-02-16-at-16-51-01.png)
![Screenshot 2022-02-16 at 16.50.47.png](https://zube.io/files/rancher/6f5cbea103c20f7ed0176b695b0b12e6-screenshot-2022-02-16-at-16-50-47.png)

Interface does not update automatically nor, after refresh, we see the move of the cluster assignment to another workspace, so the feature itself is not working at all.

Since a request is done successfully, the problem could be on the BE side, although I don't have the experience to check if the information that we send on the payload from the FE is correct or not.

Also to highlight the fact that the shouldn't be allowed to move the **local ** cluster to another workspace, which the UI allows for. That can create unforeseen problems and, imo, doesn't seem to add up.


So, to sum it up, 2 issues:
- Doing "assign to" doesn't work
- User shouldn't be able to move the "local" cluster to another workspace

Missing here a bit more knowledge to move on @nwmac . How to proceed here?

**Comment by aalves08** [comment]
Need further information. Thread here: https://suse.slack.com/archives/C02DG8ZUALR/p1645183585238339

**Comment by gaktive** [comment]
In case anything happens to Slack again, this ties to a previous question in https://github.com/rancher/rancher/issues/36132. 

For those without Slack access, there's information that the "fleetWorkspaceName" parameter on the v3 Cluster is the correct field here: https://github.com/rancher/rancher/blob/release/v2.6/pkg/apis/management.cattle.io/v3/cluster_types.go#L140

@aalves08 do we need 36132 acted on to unblock this? or what does the backend have to work on to help us?

**Comment by aalves08** [comment]
@gaktive in order to reproduce this issue and also https://github.com/rancher/dashboard/issues/4802 we need this feature to work again.

I believe that the backend needs to tackle what is specified on https://github.com/rancher/rancher/issues/36132  by @richard-cox in order to enable the feature again.

**Comment by gaktive** [comment]
Backend work pushed to 2.6.5. If that tracks with what's expected with the UI, then we'll push this out accordingly. @aalves08 update here if that's not the case.

**Comment by catherineluse** [comment]
@aalves08 Is this issue still relevant, or can it be closed?

**Comment by aalves08** [comment]
Tbh @catherineluse I think this is stale for a long time and we might close it since I don't see any action from the BE team for long time... @gaktive what do you think?

**Comment by richard-cox** [comment]
As long as the counts are validated as part of https://github.com/rancher/rancher/issues/36132 this can be closed

**Comment by gaktive** [comment]
Closing for now. The onus will be then for the backend to reopen this (or file a new ticket) once they're in a state to work on this.
